src/xrpld/rpc/handlers/LedgerEntry.cpp

```diff
@@ -806,8 +806,40 @@ parseMPToken(Json::Value const& mptJson, Json::Value& jvResult)
     }
 }
 
+static std::optional<uint256>
+parsePermissionedDomains(Json::Value const& pd, Json::Value& jvResult)
+{
+    if (pd.isString())
+    {
+        uint256 uNodeIndex;
+        if (uNodeIndex.parseHex(pd.asString()))
+            return uNodeIndex;
+
+        jvResult[jss::error] = "malformedRequest";
+        return std::nullopt;
+    }
+
+    if (!pd.isObject() || !pd.isMember(jss::account) ||
+        !pd[jss::account].isString() || !pd.isMember(jss::seq) ||
+        (pd[jss::seq].isInt() && pd[jss::seq].asInt() < 0) ||
+        (!pd[jss::seq].isInt() && !pd[jss::seq].isUInt()))
+    {
+        jvResult[jss::error] = "malformedRequest";
+        return std::nullopt;
+    }
+
+    auto const account = parseBase58<AccountID>(pd[jss::account].asString());
+    unsigned const seq = pd[jss::seq].asUInt();
+
+    if (account)
+        return keylet::permissionedDomain(*account, seq).key;
+
+    jvResult[jss::error] = "malformedRequest";
```

Comment:
```suggestion
    jvResult[jss::error] = "malformedAccount";
```

Reply:
This will break established rules for LedgerEntry.  IMO this require dedicated PR.